src/boss/bo4/unk_46E7C.c

```diff
@@ -275,7 +275,183 @@ void DopplegangerHandleDamage(DamageParam* damage, s16 step, s16 step_s) {
+    s32 j;
+    Entity* ent;
+    u8* s2;
+    u8* data;
+    PlayerDraw* plDraw;
+
+    DOPPLEGANGER.drawFlags = DRAW_COLORS;
+    plDraw = &g_PlayerDraw[8];
+
+    switch (DOPPLEGANGER.step_s) {
+    case 0:
+        DOPPLEGANGER.velocityY = 0;
+        DOPPLEGANGER.velocityX = 0;
+        if (dopStep == Dop_StatusStone) {
+            ent = &DOPPLEGANGER + 16;
+            for (j = 16; j < 64; j++, ent++) {
+                // Entity 32 appears to be EntityPlayerDissolves
+                if (ent->entityId == 32) {
+                    g_api.PlaySfx(SFX_BO4_UNK_7E6);
+                    DOPPLEGANGER.step_s = 16;
+                    return;
+                }
+            }
+        }
+        g_api.PlaySfx(SFX_BO4_UNK_7E6);
+        func_us_801C72BC();
+        func_us_801C7340();
+        DOPPLEGANGER.velocityY = FIX(-3.25);
+        func_8010E3B8(FIX(-1.25));
+        DOPPLEGANGER.ext.player.anim = 0xC0;
+        DOPPLEGANGER.rotate = 0;
+        DOPPLEGANGER.rotPivotY = 0;
+        DOPPLEGANGER.rotPivotX = 0;
+        if (damage->effects & ELEMENT_FIRE) {
+            func_80118C28(3);
+            CreateEntFactoryFromEntity(
+                g_CurrentEntity, FACTORY(BP_BLINK_WHITE, 0x4F), 0);
+            CreateEntFactoryFromEntity(g_CurrentEntity, FACTORY(BP_51, 2), 0);
+            D_us_801D3D38 = 1;
+        } else if (damage->effects & ELEMENT_THUNDER) {
+            func_80118C28(9);
+            CreateEntFactoryFromEntity(
+                g_CurrentEntity, FACTORY(BP_BLINK_WHITE, 0x59), 0);
+            CreateEntFactoryFromEntity(g_CurrentEntity, FACTORY(BP_45, 1), 0);
+            D_us_801D3D38 = 2;
+        } else if (damage->effects & ELEMENT_ICE) {
+            func_80118C28(10);
+            CreateEntFactoryFromEntity(
+                g_CurrentEntity, FACTORY(BP_BLINK_WHITE, 0x5A), 0);
+            CreateEntFactoryFromEntity(g_CurrentEntity, BP_HIT_BY_ICE, 0);
+            D_us_801D3D38 = 3;
+            DOPPLEGANGER.drawMode = DRAW_TPAGE2 | DRAW_TPAGE;
+        } else {
+            func_80118C28(1);
+            CreateEntFactoryFromEntity(
+                g_CurrentEntity, FACTORY(BP_BLINK_WHITE, 0x53), 0);
+            CreateEntFactoryFromEntity(g_CurrentEntity, FACTORY(BP_49, 5), 0);
+            D_us_801D3D38 = 0;
+        }
+        plDraw->r0 = plDraw->g0 = plDraw->b0 = plDraw->r1 = plDraw->g1 =
+            plDraw->b1 = plDraw->r2 = plDraw->g2 = plDraw->b2 = plDraw->r3 =
+                plDraw->g3 = plDraw->b3 = 128;
+        plDraw->enableColorBlend = 1;
+        DOPPLEGANGER.step_s++;
+        break;
+    case 1:
+        if (D_us_801D3D38 == 0) {
+            if (plDraw->r0 < 248) {
+                plDraw->r0++;
+            }
+            if (plDraw->b0 >= 9) {
+                plDraw->b0--;
+            }
+
+            plDraw->r3 = plDraw->r2 = plDraw->r1 = plDraw->r0;
+            plDraw->g0 = plDraw->g1 = plDraw->b1 = plDraw->g2 = plDraw->b2 =
+                plDraw->g3 = plDraw->b3 = plDraw->b0;
```

Comment:
There are instances where these fields are used in RGB consecutive order, but other cases where the B and G fields are swapped when being copied into other structs (at least one unique to PSP).

Is it possible that there are multiple definitions of this struct or the color fields are intentionally swapped some places? 

Re: 0a9da975295af665d65181abb9a843ca4ec72cc6